### PR TITLE
ENH: Implement distinct, count distinct, count on tables, and value counts

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: ibis/pandas/execution.py [hunk 1]
```diff
@@ -239,6 +245,17 @@ def execute_any_all_series(op, data, scope=None):
...
 
+@execute_node.register(ops.CountDistinct, pd.Series)
+def execute_count_distinct_series(op, data, scope=None):
+    # TODO(phillipc): Does count distinct have a mask?
+    return data.nunique()
+
+
+@execute_node.register(ops.Count, pd.DataFrame, type(None))
+def execute_count_frame(op, data, _, scope=None):
+    return len(data)
```
Comment (jreback) [inline ibis/pandas/execution.py]: shouldn't this be ``data.count()`` this is a 1 dim reduction no, e.g. return a pd.Series).

Comment (cpcloud) [inline ibis/pandas/execution.py]: No, a `Count` operation on table means "count the number of rows in the table": https://github.com/ibis-project/ibis/blob/master/ibis/expr/api.py#L1854-L1862

We could think about defining the operation you're talking about as something else that's more general like "apply reduction `f` to every column", which would return a thing that looks like a table with one row.

Comment (jreback) [inline ibis/pandas/execution.py]: i c. ok then. maybe we should add a pandas -> ibis doc

**Comment by cpcloud** [comment]
Closing in favor of a new PR that will have an additional use case implemented.

**Pull request closed without merging**